### ISSUE TITLE
: no longer a word stop

### ISSUE THREAD

**Issue by toddbruner** [vim-perl/vim-perl#326]
# Description of issue

https://stackoverflow.com/questions/73818442/vim-9-word-boundary-change-for-perl-files/73818721#73818721
excerpt copied here:

In vim 8, if you had the following string:

package Some::Package::Name;
and you placed your cursor on the S in Some::Package::Name and did a cw it would replace the text Some with whatever you typed.

In vim 9, cw now changes from S to the e in Name.

Could this  be due to a change in
/usr/share/vim/vim90/syntax/perl.vim  
or vim9?

I received  a work around in the stackoverflow post that works, but also Andy Lester encouraged me to post the issue here so you could investigate.

Thanks for the great plugin.

If your bug is not related to highlighting or folding, please feel free
to delete the text below.

# Screenshot of the issue:

![Screenshot](...)

# Sample code:

```
package Some::Package::Name;
```

# Link to vimrc

[vimrc](https://gist.github.com/toddbruner/b56f91e0994ee05a987472fb85f70cc0)

# Output of `vim --version`

```
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled May 10 2022 08:40:37)
Included patches: 1-105
Modified by team+vim@tracker.debian.org
Compiled by team+vim@tracker.debian.org
Huge version without GUI.  Features included (+) or not (-):
+acl               +file_in_path      +mouse_urxvt       -tag_any_white
+arabic            +find_in_path      +mouse_xterm       -tcl
+autocmd           +float             +multi_byte        +termguicolors
+autochdir         +folding           +multi_lang        +terminal
-autoservername    -footer            -mzscheme          +terminfo
-balloon_eval      +fork()            +netbeans_intg     +termresponse
+balloon_eval_term +gettext           +num64             +textobjects
-browse            -hangul_input      +packages          +textprop
++builtin_terms    +iconv             +path_extra        +timers
+byte_offset       +insert_expand     -perl              +title
+channel           +ipv6              +persistent_undo   -toolbar
+cindent           +job               +popupwin          +user_commands
-clientserver      +jumplist          +postscript        +vartabs
-clipboard         +keymap            +printer           +vertsplit
+cmdline_compl     +lambda            +profile           +vim9script
+cmdline_hist      +langmap           -python            +viminfo
+cmdline_info      +libcall           +python3           +virtualedit
+comments          +linebreak         +quickfix          +visual
+conceal           +lispindent        +reltime           +visualextra
+cryptv            +listcmds          +rightleft         +vreplace
+cscope            +localmap          -ruby              +wildignore
+cursorbind        -lua               +scrollbind        +wildmenu
+cursorshape       +menu              +signs             +windows
+dialog_con        +mksession         +smartindent       +writebackup
+diff              +modify_fname      +sodium            -X11
+digraphs          +mouse             -sound             -xfontset
-dnd               -mouseshape        +spell             -xim
-ebcdic            +mouse_dec         +startuptime       -xpm
+emacs_tags        +mouse_gpm         +statusline        -xsmp
+eval              -mouse_jsbterm     -sun_workshop      -xterm_clipboard
+ex_extra          +mouse_netterm     +syntax            -xterm_save
+extra_search      +mouse_sgr         +tag_binary        
-farsi             -mouse_sysmouse    -tag_old_static    
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -Wdate-time -g -O2 -ffile-prefix-map=/build/vim-dFzTGe/vim-9.0.0105=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -D_REENTRANT -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 
Linking: gcc -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -o vim -lm -ltinfo -lselinux -lsodium -lacl -lattr -lgpm -L/usr/lib/python3.10/config-3.10-x86_64-linux-gnu -lpython3.10 -lcrypt -ldl -lm -lm 
```

**Comment by petdance** [comment]
So if I'm understanding correctly:
* In vim 8, `cw` would delete `Some` from `Some::Package::Name`.
* In vim 9, `cw` would delete all of `Some::Package::Name`
* vim 9's behavior is the expected behavior and has been for 9 years.
* You don't like the behavior, and got around it by using `setlocal iskeyword-=:` so that `:` would no longer be seen as part of a word.
* I also don't like the behavior and have long had something similar in my .vimrc:
* `au FileType perl set iskeyword+=^:  " So we don't see :: separators as part of function names.`

So to me the question is: What changed on your machine between vim 8 and vim 9? Did you have something in vim 8 that was removing the : from iskeyword, and it disappeared when you went to 9?

**Comment by toddbruner** [comment]
I could be confusing causes for sure.

So I had been developing on ubuntu systems since 16.04.  I move to 18.04 and a couple of years ago I upgraded VIM to 8 manually.  Recently, I created a 22.04 VM and have been moving my dev tools/processes over to it.  Fresh install of Ubuntu, installed VIM via jonathonf/vim PPA and finally moved over my .vimrc linked above (along with .vim dir with colors).

That said, I'm not sure why my early instances did not also exhibit the behavior I did not like, maybe a plugin was "fixing" that for me?  I did not move my pathogen stuff over, and have only added a few plugins using the vim plugin methods. 

Thanks for your help.  I don't think that there is a "problem" per-se.  I just could not figure out how to correctly change to my preferred behavior.
